src/coprocessor/dag/expr/builtin_encryption.rs

```diff
@@ -31,6 +35,118 @@ const SHA224: i64 = 224;
 const SHA256: i64 = 256;
 const SHA384: i64 = 384;
 const SHA512: i64 = 512;
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum AesMode {
+    Aes128Ecb,
+    Aes192Ecb,
+    Aes256Ecb,
+    Aes128Cbc,
+    Aes192Cbc,
+    Aes256Cbc,
+}
+
+impl AesMode {
+    pub fn key_size(self) -> aes::KeySize {
+        match self {
+            AesMode::Aes128Ecb | AesMode::Aes128Cbc => aes::KeySize::KeySize128,
+            AesMode::Aes192Ecb | AesMode::Aes192Cbc => aes::KeySize::KeySize192,
+            AesMode::Aes256Ecb | AesMode::Aes256Cbc => aes::KeySize::KeySize256,
+        }
+    }
+
+    pub fn from(mode: &[u8]) -> Option<AesMode> {
```

Comment:
Why do you discard the error here? Seems we can return a not supported error.

Reply:
Handle it where it is called, because need to add not_supported to ctx.